helix-lsp/src/client.rs

```diff
@@ -1114,6 +1115,25 @@ impl Client {
         Some(self.call::<lsp::request::InlayHintRequest>(params))
     }
 
+    pub fn text_document_color_swatches(
+        &self,
+        text_document: lsp::TextDocumentIdentifier,
+        work_done_token: Option<lsp::ProgressToken>,
+    ) -> Option<impl Future<Output = Result<Value>>> {
```

Comment:
Instead of returning a `Result<Value>` and having the caller do the `serde_json::from_value`, that all can be done in this function. See https://github.com/helix-editor/helix/commit/fbc0f956b310284d609f2c00a1f4c0da6bcac165 or the signature help request above